db/version_set.cc

```diff
@@ -1125,6 +1128,22 @@ VersionSet::Finalize(Version* v)
         // consider this level
         if (compact_ok)
         {
+            size_t grooming_trigger;
+
+            // which grooming trigger point?  based upon how long
+            //  since last compaction on this level
+            //   - less than 10 minutes?
+            if (micros_now<(m_CompactionStatus[level].m_LastCompaction + (1000000 * 60 *10)))
```

Comment:
I like the thought.  The thing that bothers me is that clock is not guaranteed to move forward on platforms forced to use gettimeofday().  So micros_now could be smaller than m_LastCompaction.  Going to add test for the case and force elapsed to zero.